transformer_engine/pytorch/module/layernorm_linear.py

```diff
@@ -152,10 +161,22 @@ def forward(
                     columnwise=backward_needs_input,
                 )
 
+        ub_obj_fprop = None
+        ln_out = None
+        if ub_overlap_ag_fprop:
+            ub_obj_fprop = get_ub(ub_name + "_fprop")
+            ln_out = ub_obj_fprop.get_buffer(input_quantizer, True)
+        elif with_quantized_norm:
+            ln_out = input_quantizer.make_empty(inputmat.shape, dtype=inputmat.dtype, device="cuda")
+        else:
+            ln_out = torch.empty_like(
+                inputmat, dtype=inputmat.dtype, memory_format=torch.contiguous_format
+            )
+
         # Apply normalization
-        ln_out, mu, rsigma = apply_normalization(
+        _, mu, rsigma = apply_normalization(
             inputmat,
-            None,
+            ln_out,
```

Comment:
We prefer constructing tensors in C++ to reduce CPU overheads:
```suggestion
        ub_obj_fprop = None
        ln_out = None
        if ub_overlap_ag_fprop:
            ub_obj_fprop = get_ub(ub_name + "_fprop")
            ln_out = ub_obj_fprop.get_buffer(input_quantizer, True)

        # Apply normalization
        ln_out, mu, rsigma = apply_normalization(
            inputmat,
            ln_out,
```

Reply:
Constructing this in PyTorch is how it was before. Switching it to `None` to construct it in C++ is triggering an error:
```
RuntimeError: /home/adener/devroot/nvte-internal/transformer_engine/common/normalization/common.h:197 in function getKernel: Unavailable kernel for this normalization config.
```

This is happening with TP overlap turned off as well. Retaining the same PyTorch initialization as before works correctly.

Reply:
I see that `release_v2.0` passes `None` into `apply_normalization`: https://github.com/NVIDIA/TransformerEngine/blob/5904a80246fcd8756e0a31a0c72ca3285b7ca2e6/transformer_engine/pytorch/module/layernorm_linear.py#L158